KSystemInformer/KSystemInformer.vcxproj

```diff
@@ -138,143 +138,179 @@
     </ClCompile>
     <Link>
-      <AdditionalDependencies>ksecdd.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>ksi.lib;FltMgr.lib;ksecdd.lib;kphlib_km.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <RandomizedBaseAddress>true</RandomizedBaseAddress>
       <AdditionalOptions>/INTEGRITYCHECK /BREPRO /DEPENDENTLOADFLAG:0x800 /PDBALTPATH:%_PDB% %(AdditionalOptions)</AdditionalOptions>
       <CETCompat>true</CETCompat>
+      <AdditionalLibraryDirectories>$(OutDir);$(SolutionDir)..\kphlib\bin\$(Configuration)$(PlatformArchitecture)\;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
     </Link>
     <DriverSign>
       <FileDigestAlgorithm>sha256</FileDigestAlgorithm>
     </DriverSign>
   </ItemDefinitionGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
     <ClCompile>
-      <AdditionalIncludeDirectories>../phnt/include;../phlib/include;include;$(IntDir);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
-      <PreprocessorDefinitions>POOL_NX_OPTIN;POOL_ZERO_DOWN_LEVEL_SUPPORT;_WIN64;_AMD64_;AMD64;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <WarningLevel>Level3</WarningLevel>
+      <AdditionalIncludeDirectories>$(SolutionDir)..\phnt\include\;$(SolutionDir)..\kphlib\include\;$(ProjectDir)include\;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <PreprocessorDefinitions>TMH_FILE=%(FileName)%(Extension).tmh;POOL_NX_OPTIN;POOL_ZERO_DOWN_LEVEL_SUPPORT;_DEBUG;DEBUG;_WIN64;_AMD64_;AMD64;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <WarningLevel>Level4</WarningLevel>
       <SDLCheck>true</SDLCheck>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
       <LanguageStandard>stdcpplatest</LanguageStandard>
       <LanguageStandard_C>stdc17</LanguageStandard_C>
-      <AdditionalOptions>/kernel %(AdditionalOptions)</AdditionalOptions>
+      <AdditionalOptions>/kernel /INTEGRITYCHECK %(AdditionalOptions)</AdditionalOptions>
       <CallingConvention>StdCall</CallingConvention>
       <StringPooling>true</StringPooling>
       <SupportJustMyCode>false</SupportJustMyCode>
       <ControlFlowGuard>Guard</ControlFlowGuard>
+      <WppEnabled>true</WppEnabled>
+      <WppTraceFunction>KphTracePrint(LEVEL,EVENT,MSG,...)</WppTraceFunction>
+      <WppFileExtensions>.c.cpp.h.hpp</WppFileExtensions>
+      <WppPreserveExtensions>.c.cpp.h.hpp</WppPreserveExtensions>
+      <DisableSpecificWarnings>4201</DisableSpecificWarnings>
     </ClCompile>
     <Link>
-      <AdditionalDependencies>ksecdd.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>ksi.lib;FltMgr.lib;ksecdd.lib;kphlib_km.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <RandomizedBaseAddress>true</RandomizedBaseAddress>
       <AdditionalOptions>/INTEGRITYCHECK /BREPRO /DEPENDENTLOADFLAG:0x800 /PDBALTPATH:%_PDB% %(AdditionalOptions)</AdditionalOptions>
```

Comment:
I wanted to ask about `/PDBALTPATH:%_PDB%` option.
Is it really necessary to have it in Debug builds?

With the option there is only PDB name without full path inside driver binary. Which makes it harder to load symbols during debugging.
When full path is present, WinDbg loads symbols right away without any special action needed (otherwise I had to put it to concrete folder in my symbol cache).

Or do you know some other trick how to load symbols more easily?

Reply:
I usually just add the build output paths to `_NT_SYMBOL_PATH` so I never need to think about it again.

We could probably remove this truncation in debug builds and only have it in release builds. I'd have to look at the other `vcxproj` files too. Anyway, I think this ask it outside of the scope for this PR. If you like please open a ticket for it and we can discuss it there.

Reply:
> I usually just add the build output paths to _NT_SYMBOL_PATH so I never need to think about it again.

Yup. But that requires manual action.

> Another alternative is to add some tooling to use symstore.exe to store output of build symbols automatically in a symbol store location of the developer's choosing.

That would be nice. But why to bother with that when all that is needed is to keep full PDB paths in binary.

> We could also integrate this into our build pipeline so we can have our own public symbol store 😄.

It would be really great if you can provide symbols for public builds (symbol server would be cool, but even some package that can be downloaded will be nice).

> We could probably remove this truncation in debug builds and only have it in release builds.

Other option would be to remove the linker option from projects completely and then _force_ the setting in build scripts by setting `LINK=/PDBALTPATH:%_PDB%` environment variable.
That way developers will have full local paths in binaries (which is fine) and official builds will have stripped paths.

> If you like please open a ticket for it and we can discuss it there.

Will do.
I just dealt with this recently and since I saw this driver overhaul PR I wanted to ask about it.